### PR TITLE
Migrate `AuthenticationActivity` to use the new navigation framework

### PR DESCRIPTION
**Pull request by vinaysshenoy**
https://app.clubhouse.io/simpledotorg/story/2183/migrate-authenticationactivity-to-the-new-navigation-framework

### CODE REVIEW

**Review by msasikanth — DISMISSED** [review]
LGTM. Correct me if I am wrong, we only need to call `Router#onReady` only at Activity level since it needs to restore the router state/history right?

**Review by msasikanth — APPROVED** [review]

**Comment by vinaysshenoy** [comment]
> LGTM. Correct me if I am wrong, we only need to call `Router#onReady` only at Activity level since it needs to restore the router state/history right?

Yup, that is correct.

File: app/src/main/java/org/simple/clinic/registerorlogin/AuthenticationActiivtyComponent.kt [hunk 1]
```diff
@@ -45,7 +46,7 @@ interface AuthenticationActivityComponent :
   @Subcomponent.Builder
```
[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/registerorlogin/AuthenticationActiivtyComponent.kt:45]: Good point, will create a ticket.

[the remark this answers is not in the document]
Comment (vinaysshenoy) [reply app/src/main/java/org/simple/clinic/registerorlogin/AuthenticationActiivtyComponent.kt:45]: https://app.clubhouse.io/simpledotorg/story/2269/inject-runtime-dependencies-via-assisted-injection-when-creating-activity-injection-instances

**Pull request merged**